src/prompt/engine.go

```diff
@@ -166,6 +166,13 @@ func (e *Engine) getTitleTemplateText() string {
 }
 
 func (e *Engine) renderBlock(block *config.Block, cancelNewline bool) bool {
+	text, length := e.writeBlockSegments(block)
+
+	// do not print anything when we don't have any text
+	if length == 0 {
```

Comment:
Following the call stack - though maybe I missed something - it seems like whitespace would be significant, right? My second line had leading and trailing whitespace for aesthetics and mouse selection such that the rendered length would always be >= 2. Or was whitespace stripped if it was only whitespace somewhere? If needed, I can move any whitespace into conditionals.

Reply:
I tested this and it hides the block correctly. Whitespace segments are treated as empty.